### PR TITLE
mrc-2096 - emit click event from the control group

### PR DESCRIPTION
**Pull request by LekanAnni**
Mouse down event is not working on tree-select when triggered. If a user clicks outside the textfield or tree-select or even the label the mouse down event gets triggered, which is not ideal. This PR adds custom event(confirmEditing), which gets emitted once clickEvent occurs on tree-select or textfield.

### CODE REVIEW

File: src/DynamicFormControlSection.vue [hunk 1]
```diff
@@ -33,14 +34,15 @@
 
     import Vue from "vue";
     import DynamicFormControlGroup from "./DynamicFormControlGroup.vue";
-    import {DynamicControlGroup, DynamicControlSection} from "./types";
+    import {Control, DynamicControlGroup, DynamicControlSection} from "./types";
```
Comment (Nicolas-Dolan) [inline src/DynamicFormControlSection.vue]: Is this import being used?

File: test/dynamicForm.test.ts [hunk 2]
```diff
@@ -128,6 +129,12 @@ describe('Dynamic form component', function () {
         });
     });
 
+    it("emits confirmEditing event when triggered", async () => {
+        const rendered = getWrapper(validFormMeta, {}, mount);
+        rendered.vm.$emit("confirmEditing")
```
Comment (Nicolas-Dolan) [inline test/dynamicForm.test.ts]: I may be wrong, but should the emit not come from the DynamicFormControlSection and then you test whether DynamicForm emits in turn?

**Review by Nicolas-Dolan — CHANGES_REQUESTED** [review]
Looks good. I couldn't test it locally because there are some conflicts to be resolved. You'll need to add a test that DynamicFormControlGroup emits on click event. You'll also need to increment the version number in the package.json.

**Review by EmmaLRussell — COMMENTED** [review]
I'm not quite sure what the plan is here to fix the underlying bug, with mousedown not being propagated for the multi select controls. This PR introduces a new 'confirmEditing' event to be emitted by the form, but in HINT, the form containers handle the native mousedown event with:
` v-on:mousedown.native="confirmEditing"` so I was expecting changes to the DynamicFormMultiSelect to propagate the mousedown event. Could you explain in the PR description a bit about what's going on and how it will fix the HINT bug, and whether any changes will need to be made in HINT?

If you want to test changes to the form component in a branch of HINT, go ahead and publish a new version to npm (be sure to increment the version number). I'll add you as an editor for the package if you let me know your npm user name.

File: src/DynamicFormControlSection.vue (hunk 1 above)
Comment (LekanAnni) [reply src/DynamicFormControlSection.vue]: unused line now removed.

File: test/dynamicForm.test.ts (hunk 2 above)
Comment (LekanAnni) [reply test/dynamicForm.test.ts]: I agree,  testing has been done from DynamicFormControlSection. However, testing from Dynamic-form's parents will be done via hint as there are no implementation in this module for confirmEditing emit as I only emit custom event.

File: test/dynamicForm.test.ts [hunk 3]
```diff
@@ -145,10 +146,14 @@ describe('Dynamic form component', function () {
         });
     });
 
-    it("emits confirmEditing event when triggered", async () => {
+
+    it("emits confirmEditing event when click event triggerecd", async() => {
```
Comment (EmmaLRussell) [inline test/dynamicForm.test.ts]: This test is confusingly named as it's not the click event - maybe change to "emits confirmEditing" event when event is emitted by control section"

**Review by EmmaLRussell — COMMENTED** [review]
This looks good to me, apart from one possible test rename. Go ahead and merge when ready. Do you want to make a separate ticket for updating and testing in HINT, or are you going to use the same ticket number for the HINT branch?

**Pull request merged**